enterprise/course_catalog_api.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+"""
+Utilities to get details from the course catalog API.
+"""
+from django.utils.translation import ugettext_lazy as _
+
+try:
+    from openedx.core.djangoapps.api_admin.utils import course_discovery_api_client
+    from openedx.core.djangoapps.catalog.models import CatalogIntegration
+    from openedx.core.lib.edx_api_utils import get_edx_api_data
+except ImportError:
+    course_discovery_api_client = None
+    CatalogIntegration = None
+    get_edx_api_data = None
+
+
+class NotConnectedToOpenEdX(Exception):
+    """
+    Exception to raise when we weren't able to import needed resources.
+
+    This is raised when we try to use the resources, rather than on
+    import, so that during testing we can load the module
+    and mock out the resources we need.
+    """
+
+    pass
+
+
+def get_catalog_api_client(user):
+    """
+    Retrieve a course catalog API client.
+
+    This method retrieves an authenticated API client that can be used
+    to access the course catalog API. It raises an exception to be caught at
+    a higher level if the package doesn't have OpenEdX resources available.
+    """
+    if course_discovery_api_client is None:
+        raise NotConnectedToOpenEdX(
+            _('To get a catalog API client, this package must be installed in an OpenEdX environment.')
+        )
+    else:
+        return course_discovery_api_client(user)
+
+
+def get_all_catalogs(user):
```

Comment:
The course discovery API provides granular permissions for the catalogs; additional users or groups can be added, either on a per-catalog basis, or a by-model basis. The catalog API cannot be accessed anonymously, so the best we can do is generate an API client using a token generated with the current user and the secret OAuth token.